command/agent.go

```diff
@@ -186,32 +188,42 @@ func (c *AgentCommand) Run(args []string) int {
 	}
 
 	// Validation
-	if len(c.flagConfigs) != 1 {
-		c.UI.Error("Must specify exactly one config path using -config")
+	if len(c.flagConfigs) < 1 {
+		c.UI.Error("Must specify exactly at least one config path using -config")
 		return 1
 	}
 
-	// Load the configuration file
-	config, err := agentConfig.LoadConfig(c.flagConfigs[0])
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error loading configuration from %s: %s", c.flagConfigs[0], err))
-		return 1
+	config := agentConfig.NewConfig()
+
+	for _, configPath := range c.flagConfigs {
+		configFromPath, err := agentConfig.LoadConfig(configPath)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error loading configuration from %s: %s", configPath, err))
+			return 1
+		}
+		config = config.Merge(configFromPath)
 	}
 
 	// Ensure at least one config was found.
-	if config == nil {
+	if config == agentConfig.NewConfig() {
```

Comment:
I don't understand what this `if` is doing, could you explain please?  It seems to me that  the condition would never hold true.